src/triggers.js

```diff
@@ -722,6 +722,16 @@ function builtInTriggerValidator(options, request) {
       }
     }
   }
+  const userRoles = options.requireUserRoles;
+  if (userRoles) {
+    const roleQuery = new Parse.Query(Parse.Role);
+    roleQuery.containedIn('name', userRoles);
+    roleQuery.equalTo('users', reqUser);
+    const role = await roleQuery.first({ useMasterKey: true });
+    if (!role) {
+      throw `Validation failed. User does not match the required roles.`;
+    }
+  }
```

Comment:
May you should log the request object, because if user is authenticated you should already have user roles somewhere into the request object

Reply:
in `Auth.js` file you can find a `getUserRoles()` method
So in your use case, you could try `request.auth.getUseRoles()` and then just run an `include` on the name.
Also `getUseRoles()` has a cache behavior, so may it could be already fetched or avoid a fetch after in the cloud function.

Reply:
thank you for the pointer!